File: matlab/src/api/pfqn/pfqn_pbk.m
%{
%{
 % @file pfqn_pbk.m
 % @brief Iterative PB(k) proportional bounds (Eager-Sevcik) for single-class
 %        closed networks with delay.
%}
%}

function [Xlo,Xhi] = pfqn_pbk(L,N,Z,k)
%{
%{
 % @brief PB(k): the Eager-Sevcik proportional (performance) bound at
 %        iteration count k, i.e. the level-k member of the Performance Bound
 %        Hierarchy computed at populations N, N-1, ..., N-k (Casale-Muntz-
 %        Serazzi 2008 cite this as the iterative extension used in Tables 5,7).
 %        Reduces to the noniterative asymptotic bound at k=0 and tightens to
 %        exact as k->N. Backed by the validated PBH recursion (pfqn_pbh).
 % @fn pfqn_pbk(L, N, Z, k)
 % @param L Service demand vector (M x 1).
 % @param N Population (scalar).
 % @param Z Think time (scalar, default 0).
 % @param k Iteration count >= 0 (default 1).
 % @return Xlo Lower throughput bound.
 % @return Xhi Upper throughput bound.
%}
%}
if nargin < 3 || isempty(Z), Z = 0; end
if nargin < 4 || isempty(k), k = 1; end
[Xlo,Xhi] = pfqn_pbh(L,N,Z,k);
end

File: jar/src/main/java/jline/api/pfqn/mva/Pfqn_pbk.java
/**
 * @file Iterative PB(k) proportional bounds (Eager-Sevcik) for single-class closed networks
 *
 * PB(k) is the level-k member of the Performance Bound Hierarchy; realized through the
 * validated pfqn_pbh recursion. Ported at parity from MATLAB pfqn_pbk.m.
 *
 * @since LINE 3.0
 */
package jline.api.pfqn.mva;

import jline.util.matrix.Matrix;

public final class Pfqn_pbk {
    private Pfqn_pbk() {}

    /**
     * Iterative proportional bounds PB(k).
     *
     * @param L service demand vector (M x 1)
     * @param N population
     * @param Z think time
     * @param k iteration count (>= 0)
     * @return {Xlo, Xhi} lower/upper throughput bounds
     */
    public static double[] pfqn_pbk(Matrix L, double N, double Z, int k) {
        return Pfqn_pbh.pfqn_pbh(L, N, Z, k);
    }
}
